.github/workflows/swift-toolchain.yml

```diff
@@ -368,6 +373,8 @@ jobs:
                 -D CMAKE_MT=mt `
                 -D CMAKE_INSTALL_PREFIX=${{ github.workspace }}/BuildRoot/Library/sqlite-3.43.2/usr `
                 -D CMAKE_SYSTEM_NAME=${{ matrix.os }} `
+                -D CMAKE_ANDROID_NDK=$NDKPATH `
+                -D CMAKE_ANDROID_API=${{ needs.context.outputs.ANDROID_API_LEVEL }} `
```

Comment:
These are android specific, which can confuse the non-Android builds. Can we extract them somehow?

Reply:
If we set them conditionally (e.g. using `matrix.extra_flags`) then variables like $NDKPATH are not expanded and CMake configure fails. `CMAKE_ANDROID_API` does not suffer from this problem since it is not using powershell variables - We could move that one back to `matrix.extra_flags` but I left here for consistency. I could move that one back.

Re: confusing the build - IIUC these are ignored when `CMAKE_SYSTEM_NAME != 'Android'` based on the test builds in the PR description and on https://github.com/Kitware/CMake/blob/master/Modules/Platform/Android-Determine.cmake#L4.

Reply:
Yeah, but I'm still concerned because there are other weird behaviours with the Swift build and custom handling in the Swift modules which are not part of the CMake repository. I'd rather find a different way to handle that so that we can keep things stable.